### ISSUE TITLE
Make Grid Column and Filter Menu 508 compatible 

### ISSUE THREAD

**Issue by Alex-Bubblemaster** [telerik/kendo-ui-core#2193]
### Enhancement/Accessibility

Ticket ID: 1016555 & 1063395
### Expected/Desired behaviour

The text of the currently highlighted menu item (that is selected using the up\down arrow keys) should be read out.  If it could be prefixed with the title of the column that would be even better.

**Comment by tsvetomir** [comment]
@tapopov can you assist?

**Comment by tapopov** [comment]
@tsvetomir We need to research this thoroughly first. Although it might seem like an easy task, accessibility in general has proven to be much more complicated due to poor spec implementations by both the browsers and the major screen readers. I', sure @ggkrustev and @vladimirivanoviliev can confirm the latter. Their thoughts on the matter is also welcome.

**Comment by tapopov** [comment]
@vladimirivanoviliev and I researched the possible solutions and concluded that the easiest approach would be to allow the Menu widget to update its aria-activedescendant attribute. This can be done by assigning the ColumnMenu an unique ID (for example the column's field property followed by some suffix). The results are very similar to what the standalone Menu widget would produce - the screen readers simply read out the text of the focused Menu item. Here is ChromeVox output example: 

```javascript
// Move focus to Grid's column header
Speak (I) "Column Settings Last Name has popup"

// Press ALT+Down to open
Speak (I) category=nav "Last Name"
Speak (Q) category=nav "Internal link"
Speak (I) category=nav "Exited Grid has popup Column header has popup."
Speak (Q) category=nav ""
Speak (Q) category=nav "Sort Ascending"
Speak (Q) category=nav "Menu bar"

// Press Down to move focused item
Speak (I) "Sort Descending"

// Press Down to move focused item
Speak (I) "Columns"

// Press Down to move focused item
Speak (I) "Filter"
```

@ggkrustev, @tsvetomir what is your opinion? Is the information we provide enough or should we provide more context to the users? Do we need to associate each Menu item with the Grid's column header? I am not sure that we do and furthermore, doing this will complicate the solution immensely.

**Comment by tsvetomir** [comment]
Looks usable to me. What about the empty entry `Speak (Q) category=nav ""`?

**Comment by tapopov** [comment]
Not sure what's causing it, however it's not read out loud anyways. We've tested the same approach with JAWS and NVDA and both seem to behave well. Adding aria-checked to menu items with checkboxes is also a good idea and should be fairly easy to implement.

**Comment by ggkrustev** [comment]
The chosen approach looks good. The reader speaks out pretty reasonable information that should be sufficient. 

Adding an `aria-checked` will be beneficial. Just consider using a proper **role** (probably [`menuitemcheckbox`](https://www.w3.org/TR/wai-aria/roles#menuitemcheckbox)), otherwise the aria attribute will be invalid.

I would like only to suggest using an unique IDs (guids) instead of column name + suffix. There could be more than one column with same name on the page.

**Comment by tapopov** [comment]
Done and merged in production.

**Comment by tapopov** [comment]
While the column menus and headers are being read out as expected, the filter menu itself is not.

**Comment by tsvetomir** [comment]
The FilterMenu should be fixed 2017.1.118
